paimon-core/src/main/java/org/apache/paimon/utils/SnapshotManager.java

```diff
@@ -253,6 +256,17 @@ public List<Snapshot> safelyGetAllSnapshots() throws IOException {
                         .map(this::snapshotPath)
                         .collect(Collectors.toList());
 
```

Comment:
Maybe we can enrich the `BranchManager::branches()` to get all the branch set info? You can rebase master to see the implementation.

Reply:
Good idea！I will fix it.